mkosi/state.py

```diff
@@ -21,6 +24,15 @@ def __init__(self, args: MkosiArgs, config: MkosiConfig, workspace: Path) -> Non
         self.pkgmngr.mkdir()
         self.install_dir.mkdir(exist_ok=True)
         self.cache_dir.mkdir(parents=True, exist_ok=True)
+        source_date_epoch = self.config.environment.get("SOURCE_DATE_EPOCH")
+        if source_date_epoch is not None:
+            try:
+                mtime_epoch = int(source_date_epoch)
+            except ValueError:
+                die(f"SOURCE_DATE_EPOCH={source_date_epoch} is not a valid integer")
+            if mtime_epoch < 0:
+                die(f"SOURCE_DATE_EPOCH={source_date_epoch} is negative")
```

Comment:
Could we maybe just (re)raise a `ValueError` in both cases? We `die` all over the place, but in a constructor it's a bit icky.